cmake/LTO.cmake

```diff
@@ -0,0 +1,57 @@
+
+set (MONGO_LTO "OFF"
```

Comment:
I feel like the variable should actually be named `MONGOCRYPT_LTO`.  This brings it more into line with `MONGOCRYPT_CRYPTO`, which seems to establish the pattern of using `MONGOCRYPT_` as the prefix.  Additionally, the name `MONGO_LTO` (at least to me) implies the server project or a more generic standard variable that would apply to multiple MongoDB projects.

Reply:
This is actually chosen deliberately to match what is used in `Sanitizers.cmake` etc (currently in mongo-c-driver, but I plan on bringing that one over soon as well)